### PR TITLE
Use correct secret when creating client for Dynatrace API

### PR DESCRIPTION
**Pull request by lrgar**

### CODE REVIEW

File: pkg/controller/oneagent-utils/verification_test.go [hunk 1]
```diff
@@ -62,3 +65,29 @@ func TestVerifySecret(t *testing.T) {
 		assert.NoError(t, err)
 	}
 }
+
+func TestBuildDynatraceClientWithCustomToken(t *testing.T) {
```
Comment (baichinger) [inline pkg/controller/oneagent-utils/verification_test.go]: Suggest to add an additional test for covering the erroneous use case.

Comment (lrgar) [reply pkg/controller/oneagent-utils/verification_test.go]: This is the erroneous use case, but I can add one for the normal one (i.e., token and secret have the same name.)

Comment (baichinger) [reply pkg/controller/oneagent-utils/verification_test.go]: This one asserts `NoError`. I would add one which asserts an `Error` to verify if a faulty configuration gets handled accordingly.

Comment (lrgar) [reply pkg/controller/oneagent-utils/verification_test.go]: Ah, ok, fair, will do.

**Pull request merged**